kotlinx-coroutines-core/common/src/channels/Channels.common.kt

```diff
@@ -162,9 +165,16 @@ public suspend inline fun <E> ReceiveChannel<E>.consumeEach(action: (E) -> Unit)
     }
 
 /**
- * Returns a [List] containing all the elements sent to this channel, preserving their order.
+ * [Consumes][consume] the elements of this channel into a list, preserving their order.
  *
- * This function will attempt to receive elements and put them into the list until the channel is
+ * This is a convenience function equivalent to calling [consumeAsFlow] followed by [kotlinx.coroutines.flow.toList].
+ * It is useful for testing code that uses channels to observe the elements the channel contains at the end of the test.
```

Comment:
```suggestion
 * It is useful for testing to observe the elements the channel contains at the end of the test.
```

Reply:
I'd leave the "code that uses channels" phrasing so that people whose code doesn't use channels can easily understand that this is not the function they're looking for.